streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -3489,8 +3489,8 @@ final class ZStream[-R, +E, +A] private (val channel: ZChannel[R, Any, Any, Any,
     channel.toPull.map { pull =>
       pull.foldZIO(
         success = {
-          case Left(done)  => Exit.failNone
-          case Right(elem) => Exit.succeed(elem)
+          case Right(elem)            => Exit.succeed(elem)
+          case _: Left[Any, Chunk[A]] => Exit.failNone
```

Comment:
I don't think we need the typecheck here do we?

Reply:
We don't :)